### PR TITLE
Completed typescript-conditionals Exercise

### PR DESCRIPTION
**Pull request by Sam-Castaneda13**
Screenshot 2024-07-10 143452

### CODE REVIEW

File: typescript-conditionals/main.ts [hunk 1]
```diff
@@ -0,0 +1,161 @@
...
+
+function isOldEnoughToDriveAndDrive(person: Person): boolean {
+  if (person.age > 16) {
+    return false;
+  } else if (person.age > 20) {
+    return false;
+  } else {
+    return false;
+  }
+}
```
Comment (brandon-moy) [inline typescript-conditionals/main.ts]: Rather than individual checks for able to drive and able to drink, you should combine these into one condition that checks if both are `true`.

File: typescript-conditionals/main.ts [hunk 2]
```diff
@@ -0,0 +1,161 @@
...
+
+console.log('The Value of isOldEnoughToDrinkResults:', canDrinkChecker);
+
+function isOldEnoughToDrive(person: Person): boolean {
+  if (person.age > 16) {
+    return true;
+  } else {
+    return false;
+  }
+}
```
Comment (brandon-moy) [inline typescript-conditionals/main.ts]: A person is old enough to drive if they are _at least_ 16 years of age. Should this condition be just greater than 16?

File: typescript-conditionals/main.ts [hunk 3]
```diff
@@ -0,0 +1,161 @@
...
+  age: number;
+}
+
+function isOldEnoughToDrink(person: Person): boolean {
+  if (person.age > 21) {
+    return true;
+  } else {
+    return false;
+  }
+}
```
Comment (brandon-moy) [inline typescript-conditionals/main.ts]: A person is old enough to drink if they are _at least_ 21 years of age. Should this condition be just greater than 16?

File: typescript-conditionals/main.ts [hunk 4]
```diff
@@ -0,0 +1,161 @@
...
+  'The Value of isOldEnoughToDriveAndDrinkResults:',
+  canDriveAndDrink
+);
+
+function categorizeAcidity(pH: number): string {
+  if (pH < 7) {
+    return 'acid';
+  } else if (pH === 7) {
+    return 'neutral';
+  } else if (pH > 7 && pH < 15) {
```
Comment (brandon-moy) [inline typescript-conditionals/main.ts]: You only have if the `pH < 7` but what about negative numbers? Should be me a limit that we are checking between for the acidity?

File: typescript-conditionals/main.ts [hunk 5]
```diff
@@ -0,0 +1,161 @@
...
+      break;
+    case 'musical':
+      return 'You should watch In The Heights';
+      break;
+    case 'sci-fi':
+      return 'You should watch Star-Wars';
+      break;
+    default:
+      return 'Genre not recognized. Choose between action, comedy, horror, drama, musical, or sci-fi.';
+      break;
```
Comment (brandon-moy) [inline typescript-conditionals/main.ts]: You don't need both `return` and `break`. `return` will return a value from a function and then exit the function (which stops the execution of future code. The `break` statement will terminate the loop or switch statement but your `break` will never be reached due to your `return`.

tl;dr - you should remove the `break` statements.

File: typescript-conditionals/main.ts [hunk 6]
```diff
@@ -0,0 +1,161 @@
...
+    case 'wakko':
+      return "We're the warner brothers!";
+      break;
+    case 'dot':
+      return "I'm cute~";
+      break;
+    default:
+      return 'Goodnight everybody!';
+      break;
+  }
```
Comment (brandon-moy) [inline typescript-conditionals/main.ts]: There is a way to make different `case` statements evaluate into the same expression. I would recommend doing that here for `yakko` and `wakko` where they evaluate to the same `return "We're the warner brothers!"`

Also you do not need both `return` and `break`. Further explanation below.

**Review by brandon-moy — CHANGES_REQUESTED** [review]
Great start! Have some changes requested for this exercise.

**Review by brandon-moy — APPROVED** [review]
Great job, approved! 👍
